pkg/utils/aws.go

```diff
@@ -0,0 +1,85 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/openshift/osdctl/pkg/provider/aws"
+)
+
+type cloudCredentialsResponse struct {
+	// ClusterID
+	ClusterID string `json:"clusterID"`
+
+	// Link to the console, optional
+	ConsoleLink *string `json:"consoleLink,omitempty"`
+
+	// Cloud credentials, optional
+	Credentials *string `json:"credentials,omitempty"`
+
+	// Region, optional
+	Region *string `json:"region,omitempty"`
+}
+
+type awsCredentialsResponse struct {
+	AccessKeyId     string `json:"AccessKeyId" yaml:"AccessKeyId"`
+	SecretAccessKey string `json:"SecretAccessKey" yaml:"SecretAccessKey"`
+	SessionToken    string `json:"SessionToken" yaml:"SessionToken"`
+	Region          string `json:"Region" yaml:"Region"`
+	Expiration      string `json:"Expiration" yaml:"Expiration"`
+}
+
+// Creates an AWS client based on a clusterid
+// Requires previous log on to the correct api server via ocm login
+// and tunneling to the backplane
+func CreateAWSClient(clusterID string) (aws.Client, error) {
```

Comment:
Given this relies on getting AWS info from a cluster, I think the function name should change to reflect that. There's already an AWS package in osdctl that provides creating a raw AWS client `awsprovider.NewAwsClientWithInput` so to have two functions with similar names and no context may get confusing. At a larger level, moving this util package to something named maybe `clusterAWS` or `clusterCloud` or something along those lines may help clarify.

Reply:
Makes sense, wasn't sure where to put this. 
Moved to a new clustercloud package.